### PR TITLE
allow multiple spaces between alias and flag

### PR DESCRIPTION
**Pull request by mklabs**
Hi Sindre,

Here's a little patch to parse alias / flags with help output like:

```
-h,   --help          # Print the generator's options and usage
```

I was trying to get options / arguments from yeoman generator's help output.

Thanks a lot for this little lib, simple but very useful.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 'use strict';
 var execall = require('execall');
-var re = /^\s*[^$]\s*(?:-([a-z-]), )?--([a-z-]+) +(.*)$/igm;
+var re = /^\s*[^$]\s*(?:-([a-z-]),\s+)?--([a-z-]+) +(.*)$/igm;
```
Comment (sindresorhus) [inline index.js]: `\s` matches newlines too though, is that really wanted?

**Comment by sindresorhus** [comment]
Oh, hey Mickael. Haven't seen you in like forever.

**Comment by mklabs** [comment]
I think this is my first pull request since like, well, forever :p

File: index.js (hunk 1 above)
Comment (mklabs) [inline index.js]: oh, yeah, didn't thought about that, because of the `m` flag right ?

Is there another way ?

Comment (sindresorhus) [inline index.js]: I think `\s` always matches newlines:

> `\s` - Matches a single white space character, including space, tab, form feed, line feed and other Unicode spaces. Equivalent to [ \f\n\r\t\v​\u00a0\u1680​\u180e\u2000​-\u200a​\u2028\u2029\u202f\u205f​\u3000\ufeff]. - MDN

## 

`+` works, or you could use the space escape explicitly `\u0020+`

Comment (mklabs) [inline index.js]: Either way, what do you prefer ?

The intent here is to match both space and tab I guess.

Something like `[ \t]+` ?

Comment (sindresorhus) [inline index.js]: > Something like [ \t]+ ?

👍

Comment (mklabs) [inline index.js]: alright, pushed the update to the regexp

**Comment by sindresorhus** [comment]
Thanks :)

**Comment by mklabs** [comment]
tab

from

```
Usage:
  yo trails:model [options] <model-name>

Scaffolds a new model

Options:
  -h,   --help          # Print the generator's options and usage
        --skip-cache    # Do not remember prompt answers             Default: false
        --skip-install  # Do not automatically install dependencies  Default: false

Arguments:
  model-name    Type: String  Required: true
```

parse-help very helpful :=)

Would you consider the ability to parse commands as well ? In most case, when cli tools define commands, the help output is something like:

```
Options:
  -h, --help              Lorem Ipsum

Commands:
  install                 Blah Blah
```

This way, I could automate the completion for pretty much any program that follows this help output.

**Comment by sindresorhus** [comment]
Yes. That would be useful. PR welcome :)

**Pull request merged**